File: euler009-proc.php
#!/usr/bin/env php
<?php
/** Problem 9: Special Pythagorean triplet
 * https://projecteuler.net/problem=9
 */

function euler009(int $sum): int
{
    for ($a = 1; $a <= $sum - (2 * $a + 3); $a++) {
        for ($b = $a + 1; $b <= $sum - ($b + 1); $b++) {
            $c = $sum - ($a + $b);
            if ($a ** 2 + $b ** 2 == $c ** 2) {
                // The problem statement guarantees that this will be reached
                return $a * $b * $c;
            }
        }
    }
}

// Main
echo euler009(intval(trim(file_get_contents('php://STDIN')))) . PHP_EOL;

File: euler009-proc.py
#!/usr/bin/env python3

# Problem 9: Special Pythagorean triplet
# https://projecteuler.net/problem=9

import sys

def euler009(total):
    a = 1
    while a <= total - (2 * a + 3):
        b = a + 1
        while b <= total - (b + 1):
            c = total - (a + b)
            if a ** 2 + b ** 2 == c ** 2:
                return a * b * c
            b += 1
        a += 1

def parse_input(lines):
    return int(lines[0].strip())

if __name__ == "__main__":
    print(euler009(parse_input(sys.stdin.readlines())))
